### DISCUSSION TITLE
Use k samples for training before evaluating the progressing validation score

### DISCUSSION THREAD

**Discussion by SirPopiel** [online-ml/river#756] [Q&A] [answered]
Hello, I was wondering if it were easily possible to use k samples (say 100) of a data stream for training before evaluating the progressing validation score in an online fashion. It would be useful for me, as I'm using River's algorithms to benchmark the results of some models which assume to have (a few) training data before starting to work online.
Thank you in advance!

**Comment by MaxHalford** [comment, marked as answer]
Hello! What's preventing you from doing that right now? For example:

```py
import itertools
from river import datasets
from river import evaluate
from river import linear_model
from river import metrics
from river import preprocessing

model = (
    preprocessing.StandardScaler() |
    linear_model.LogisticRegression()
)

dataset = iter(datasets.Phishing())

# Warm up
k = 200
for x, y in itertools.islice(dataset, k):
    model.predict_one(x)
    model.learn_one(x, y)

evaluate.progressive_val_score(
    model=model,
    dataset=dataset,
    metric=metrics.ROCAUC(),
    print_every=100
)
```

**Reply by SirPopiel** [reply]
Yup, you are right, my bad. Thank you!

**Reply by MaxHalford** [reply]
No worries! Never hesitate to ask.
